src/test/kotlin/org/rust/ide/inspections/RsLivenessInspectionTest.kt

```diff
@@ -489,9 +489,30 @@ class RsLivenessInspectionTest : RsInspectionsTestBase(RsLivenessInspection::cla
         macro_rules! my_macro {
             ($ e:expr) => { unresolved!(e) };
         }
-        
+
         fn foo(x: i32) {
             my_macro!(x);
         }
     """)
+
+    fun `test destructured struct variable`() = checkByText("""
+        struct S {
+            a: u32
+        }
+
+        fn foo() {
+            let S { a } = S { a: 0 };
+            a;
+        }
+    """)
+
+    fun `test destructured struct parameter`() = checkByText("""
```

Comment:
Please also ensure that destructured struct parameters work fine in move analysis and add such tests to `RsBorrowCheckerMovesTest`